mmpy_bot/function.py

```diff
@@ -92,12 +92,9 @@ def __init__(
                 info_name=self.matcher.pattern.strip("^").split(" (.*)?")[0],
             ) as ctx:
                 # Get click help string and do some extra formatting
-                self.docstring = self.function.get_help(ctx).replace(
-                    "\n", f"\n{spaces(8)}"
-                )
+                self.docstring += f"\n\n{self.function.get_help(ctx)}"
```

Comment:
I think this is what causes those duplicate docstrings you showed earlier in Discord. `click` already includes the docstring in `get_help`, so this shouldn't be `+=`.

```suggestion
                self.docstring = f"\n\n{self.function.get_help(ctx)}"
```

Reply:
I tried `=` before but then we lose all of the help from `click`.
I'm not happy with either of the solutions but I think we'll need to address `click` UI/UX in another PR.
I agree that `click` is convenient for argument parsing but the help information is (IMO) too verbose for a chat-bot.

Reply:
> I tried = before but then we lose all of the help from click.

What do you mean? The "help from click" is what is returned by `self.function.get_help(ctx)` right?

Reply:
Yes to `self.function.get_help(ctx)`. If memory serves the issue was functions decorated with both `@listen_to` and `@click...`. The inner click decorated is processed and `function.get_help(ctx)` is called but as the outer decorator (`listen_to`) is processed later, this was overwriting what was done before, hence the `+=`.